src/app/routes/getInitialData/index.js

```diff
@@ -8,10 +9,19 @@ const upstreamStatusCodesToPropagate = [200, 404];
 const getInitialData = async ({ match }) => {
   const { id, service, amp } = match.params;
   const isAmp = !!amp;
-  const url = `${process.env.SIMORGH_BASE_URL}/${service}/articles/${id}.json`;
-
   let data;
   let status;
+  let baseUrl = process.env.SIMORGH_BASE_URL;
+
+  if (
+    typeof window !== 'undefined' &&
+    window.location &&
+    window.location.origin
+  ) {
+    baseUrl = getBaseUrl(window.location.origin);
+  }
```

Comment:
Think this would be cleaner if `getBaseUrl` handled both cases. So in this file:

```js
const baseUrl = getBaseUrl();
```

In the other file:

```js
const getBaseUrl = () => {
  const isClient = (window && window.location && window.location.origin);
  const origin = isClient ? window.location.origin : process.env.SIMORGH_BASE_URL;
  ...etc
};
```

Reply:
I extracted out its its own function so it could be better unit tested, as we use jsdom, dealing with window makes simple input outputs test more complicated